cmd/unused-exporter/exporter.go

```diff
@@ -158,14 +170,61 @@ func (e *exporter) Collect(ch chan<- prometheus.Metric) {
 				di.Count += 1
 				di.SizeByType[d.DiskType()] += int64(d.SizeGB())
 			}
+
+			var ms []metric // TODO we can optimize this creation here and allocate memory only once
+
+			emit := func(d *prometheus.Desc, v int64, lbls ...string) {
```

Comment:
Two small NITs here

Any reason this function has to be defined inline?  I think it could be pulled out to be simply a re-usable function.  Not sure what the go best practice is here, but it'd at least break up this long block.

Other thing is, can we call this something other than `emit`?  Really it's not _emitting_ anything, its writing it to the cache that the collector reads from.  Maybe something like `cacheResults()`?

Reply:
> Any reason this function has to be defined inline? I think it could be pulled out to be simply a re-usable function. Not sure what the go best practice is here, but it'd at least break up this long block.

The only reason is it because it's a closure that uses variables defined outside this function (`ms`, `providerName`, and `providerID`). Extracting to a function would mean having to pass these as arguments to the function, making not only overtly verbose but also very repetitive.

> Other thing is, can we call this something other than `emit`? Really it's not _emitting_ anything, its writing it to the cache that the collector reads from. Maybe something like `cacheResults()`?

The name is a remnant of the previous version, where we were _emitting_ the metric to the channel. I don't think `cacheResults` is more descriptive, maybe something like `addMetric` instead? 🤔

Reply:
> The only reason is it because it's a closure that uses variables defined outside this function (ms, providerName, and providerID). Extracting to a function would mean having to pass these as arguments to the function, making not only overtly verbose but also very repetitive.

Makes sense

> The name is a remnant of the previous version, where we were emitting the metric to the channel. I don't think cacheResults is more descriptive, maybe something like addMetric instead? 🤔

`addMetric` works for me.